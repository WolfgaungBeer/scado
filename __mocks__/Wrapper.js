import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
