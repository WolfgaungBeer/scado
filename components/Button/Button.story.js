import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import Button from './';
import LabeledIcon from '../LabeledIcon';


storiesOf('Button', module)
    .add('default', () => (
        <Button onClick={action('clicked')}>Click me!</Button>
    ))
    .add('with props', () => (
        <Button
            type={text('type', 'flat')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            onClick={action('clicked')}
        >
            Click me!
        </Button>
    ))
    .add('with a custom component', () => (
        <Button
            type={text('type', 'flat')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            onClick={action('clicked')}
        >
            <LabeledIcon icon="dashboard" label="CLICK ME" />
        </Button>
    ));
