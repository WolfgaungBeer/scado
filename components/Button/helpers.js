export const getFontColor = () => (props) => {
    switch (props.type) {
    case 'flat':
        return props.color ? props.theme.color[props.color] : props.theme.color.black;
    case 'raised':
        return props.color ? props.theme.color.white : props.theme.color.black;
    default:
        return props.color ? props.theme.color.white : props.theme.color.black;
    }
};

export const getColor = () => (props) => {
    if (props.type) {
        return (props.type === 'flat') ? 'transparent' : props.theme.color[props.color];
    }
    return props.theme.color[props.color] || props.theme.color[props.lightGray];
};

export const getBorderStyle = () => (props) => {
    if (props.type) {
        return (props.type === 'flat') ? 'none' : 'solid';
    }
    return 'solid' || 'none';
};

export const getIconColor = (type, color) => {
    switch (type) {
    case 'flat':
        return color || 'black';
    case 'raised':
        return color ? 'white' : 'black';
    default:
        return color ? 'white' : 'black';
    }
};
