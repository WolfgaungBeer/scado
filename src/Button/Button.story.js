import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
import Button from './';
import Icon from '../Icon';
import Text from '../Text';

storiesOf('Button', module)
    .add('default', () => (
        <Button onClick={action('clicked')}>Click me!</Button>
    ))
    .add('with props', () => (
        <Button
            flat={boolean('flat', false)}
            raised={boolean('raised', false)}
            border={boolean('border', false)}
            disabled={boolean('disabled', false)}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            onClick={action('clicked')}
        >
            Click me!
        </Button>
    ))
    .add('with a custom component', () => (
        <Button raised color="error" onClick={action('clicked')}>
            <div>
                <Icon color="white" scale="xl">dashboard</Icon>
                <Text.Span color="white" scale="xl">Das ist ein Text</Text.Span>
            </div>
        </Button>
    ));
