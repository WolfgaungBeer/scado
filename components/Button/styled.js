import styled from 'styled-components';
import { getBackgroundColor, getBorder, getShadow, getFontColor } from './helpers';

export const StyledButton = styled.button`
    width: 100%;
    cursor: pointer;
    background-color: ${getBackgroundColor()};
    ${getBorder()};
    ${getShadow()};
    ${getFontColor()}
    font-size: ${props => props.theme.scale[props.scale]};
    padding: 0.5rem 1rem;
    &:focus {
        outline: 0;
    }
`;
