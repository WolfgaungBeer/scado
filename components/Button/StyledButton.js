import styled from 'styled-components';
import { getFontColor, getColor, getBorderStyle } from './helpers';

const StyledButton = styled.button`
    cursor: pointer;
    font-size: ${props => props.theme.scale[props.scale]};
    color: ${getFontColor()};
    background-color: ${getColor()};
    border-style: ${getBorderStyle()};
    border-color: ${getColor()};
    border-width: 1px;
    border-radius: 2px;
    box-shadow: ${props => ((props.type === 'raised') ? '3px 3px 10px -3px rgba(0,0,0,0.75)' : '')};
    &:focus {
        outline: 0;
    }
    > div {
        padding: ${props => (props.iconOnly ? '0.25rem' : '0.25rem 0.5rem')};
        > i {
            transform: translateY(13%);
            padding-right: ${props => (props.iconOnly ? '' : '0.25rem')};
        }
    }
`;

StyledButton.displayName = 'StyledButton';

export default StyledButton;
