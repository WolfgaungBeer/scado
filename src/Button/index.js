import React from 'react';
import { node, func, oneOf, string, bool } from 'prop-types';
import { StyledButton } from './styled';

const propTypes = {
    type: oneOf(['submit', 'button']),
    flat: bool,
    raised: bool,
    border: bool,
    disabled: bool,
    color: string,
    scale: string,
    children: node,
    onClick: func,
};

const defaultProps = {
    type: undefined,
    flat: undefined,
    raised: undefined,
    border: undefined,
    disabled: undefined,
    color: undefined,
    scale: undefined,
    children: undefined,
    onClick: undefined,
};

const Button = ({ type, flat, raised, border, disabled, color, scale, children, onClick }) => (
    <StyledButton
        type={type}
        flat={flat}
        raised={raised}
        border={border}
        disabled={disabled}
        color={color}
        scale={scale}
        onClick={onClick}
    >
        {children}
    </StyledButton>
);

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
