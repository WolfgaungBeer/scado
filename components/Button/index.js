import React from 'react';
import { string, func, node } from 'prop-types';
import StyledButton from './StyledButton';
import { getIconColor } from './helpers';
import Icon from '../Icon';


const propTypes = {
    type: string,
    color: string,
    scale: string,
    icon: string,
    onClick: func,
    children: node
};

const defaultProps = {
    type: undefined,
    color: undefined,
    scale: undefined,
    icon: undefined,
    onClick: undefined,
    children: undefined
};

const Button = ({ type, color, scale, icon, onClick, children }) => {

    const buttonType = type || 'default';
    const iconColor = getIconColor(type, color);
    const iconOnly = (icon && !children);

    return (
        <StyledButton type={buttonType} color={color} scale={scale} iconOnly={iconOnly} onClick={onClick}>
            <div>
                {icon && <Icon icon={icon} color={iconColor} scale={scale} />}
                {children}
            </div>
        </StyledButton>
    );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
