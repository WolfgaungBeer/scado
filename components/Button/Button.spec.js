import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Button from './';

const click = () => {};

it('renders the Button Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Button onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Button color="error" scale="l" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Button buttonType="flat" color="error" scale="l" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();

    const tree4 = renderer.create(
        <Wrapper>
            <Button buttonType="border" color="error" scale="l" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree4).toMatchSnapshot();

    const tree5 = renderer.create(
        <Wrapper>
            <Button buttonType="raised" color="error" scale="l" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree5).toMatchSnapshot();

    const tree6 = renderer.create(
        <Wrapper>
            <Button buttonType="flat" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree6).toMatchSnapshot();

    const tree7 = renderer.create(
        <Wrapper>
            <Button buttonType="border" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree7).toMatchSnapshot();

    const tree8 = renderer.create(
        <Wrapper>
            <Button buttonType="raised" onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree8).toMatchSnapshot();

    const tree9 = renderer.create(
        <Wrapper>
            <Button buttonType="raised" disabled onClick={click}>TEST</Button>
        </Wrapper>,
    ).toJSON();
    expect(tree9).toMatchSnapshot();
});
