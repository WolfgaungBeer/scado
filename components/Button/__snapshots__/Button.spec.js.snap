// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Button Component correctly 1`] = `
<button
  className="sc-chPdSV poBNe"
  color={undefined}
  disabled={undefined}
  onClick={[Function]}
  scale={undefined}
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 2`] = `
<button
  className="sc-chPdSV cWtCRk"
  color="error"
  disabled={undefined}
  onClick={[Function]}
  scale="l"
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 3`] = `
<button
  className="sc-chPdSV fQbBaF"
  color="error"
  disabled={undefined}
  onClick={[Function]}
  scale="l"
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 4`] = `
<button
  className="sc-chPdSV dDMJZi"
  color="error"
  disabled={undefined}
  onClick={[Function]}
  scale="l"
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 5`] = `
<button
  className="sc-chPdSV ejMXTh"
  color="error"
  disabled={undefined}
  onClick={[Function]}
  scale="l"
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 6`] = `
<button
  className="sc-chPdSV cINVmk"
  color={undefined}
  disabled={undefined}
  onClick={[Function]}
  scale={undefined}
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 7`] = `
<button
  className="sc-chPdSV iraiXO"
  color={undefined}
  disabled={undefined}
  onClick={[Function]}
  scale={undefined}
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 8`] = `
<button
  className="sc-chPdSV kysexW"
  color={undefined}
  disabled={undefined}
  onClick={[Function]}
  scale={undefined}
  type={undefined}
>
  TEST
</button>
`;

exports[`renders the Button Component correctly 9`] = `
<button
  className="sc-chPdSV iBKnuX"
  color={undefined}
  disabled={true}
  onClick={[Function]}
  scale={undefined}
  type={undefined}
>
  TEST
</button>
`;
